python/ray/tune/trial_runner.py

```diff
@@ -396,88 +398,66 @@ def add_trial(self, trial):
-            else:
-                for t in sorted_trials:
-                    messages.append(" - {}:\t{}".format(
-                        t, t.progress_string()))
+    def debug_string(self,
+                     parameters=None,
+                     metrics=None,
+                     format="psql",
+                     max_rows=100):
+        """Returns a human readable message for printing to the console.
 
-        return "\n".join(messages) + "\n"
+        This contains a table where each row represents a trial, its parameters
+        and the current values of its metrics.
 
-    def _debug_messages(self):
+        Args:
+            parameters (List[str]): Names of trial parameters to include.
+                Defaults to all parameters across all trials.
+            metrics (List[str]): Names of metrics to include. Defaults to
+                metrics defined in DEFAULT_RESULT_KEYS.
+            format (str): Output format (see tablefmt in tabulate API).
+            max_rows (int): Maximum number of rows in the trial table.
+        """
         messages = ["== Status =="]
         messages.append(self._scheduler_alg.debug_string())
         messages.append(self.trial_executor.debug_string())
-        messages.append(self._memory_debug_string())
-        return messages
-
-    def _memory_debug_string(self):
-        try:
-            import psutil
-            total_gb = psutil.virtual_memory().total / (1024**3)
-            used_gb = total_gb - psutil.virtual_memory().available / (1024**3)
-            if used_gb > total_gb * 0.9:
-                warn = (": ***LOW MEMORY*** less than 10% of the memory on "
-                        "this node is available for use. This can cause "
-                        "unexpected crashes. Consider "
-                        "reducing the memory used by your application "
-                        "or reducing the Ray object store size by setting "
-                        "`object_store_memory` when calling `ray.init`.")
-            else:
-                warn = ""
-            return "Memory usage on this node: {}/{} GiB{}".format(
-                round(used_gb, 1), round(total_gb, 1), warn)
-        except ImportError:
-            return ("Unknown memory usage. Please run `pip install psutil` "
-                    "(or ray[debug]) to resolve)")
+        messages.append(memory_debug_string())
+
+        delim = "<br>" if format == "html" else "\n"
+        trials = self.get_trials()
+
+        if len(trials) < 1:
+            return delim.join(messages) + delim
+
+        num_trials = len(trials)
+        num_trials_per_state = collections.defaultdict(int)
+        for t in trials:
+            num_trials_per_state[t.status] += 1
+        messages.append("Number of trials: {} ({})"
+                        "".format(num_trials, num_trials_per_state))
+        if num_trials > max_rows:
+            messages.append("Table truncated to {} rows.".format(max_rows))
```

Comment:
Maybe indicate number of overflow; and also indicate a todo for a suggestion for users to view more?